src/test/java/strategy/BalanceOperationStrategyTest.java

```diff
@@ -0,0 +1,39 @@
+package strategy;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import dao.FruitStorageDao;
+import dao.FruitStorageDaoImpl;
+import java.util.HashMap;
+import java.util.Map;
+import model.Fruit;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class BalanceOperationStrategyTest {
+    private FruitStorageDao fruitStorageDao;
+    private BalanceOperationStrategy balanceOperationStrategy;
+
+    @BeforeEach
+    void setUp() {
+        fruitStorageDao = new FruitStorageDaoImpl();
+        balanceOperationStrategy = new BalanceOperationStrategy();
+        fruitStorageDao.set("banana", 15);
+        fruitStorageDao.set("apple", 10);
+    }
+
+    @Test
+    void handleBalanceOperation_ok() {
+        Map<String,Integer> expected = new HashMap<>();
```

Comment:
what happens if we try to set negative values?

Reply:
it will except the negative value, but after that FruitStorageDaoImpl will throw an exception that q-ty should be greater or equal to zero..